### ISSUE TITLE
Error importing values from JSON files if they're called "tsconfig.json"

### ISSUE THREAD

**Issue by maranomynet** [oven-sh/bun#8065]
### What version of Bun is running?

1.0.21

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

Create the following files:

`./tsconfig.json`:
```json
{ "extends": 1 }
```

`./test.mjs`
```ts
import test from './tsconfig.json' assert { type: "json" };
console.log(test);
```

Run the script: 
```sh
$ bun test.mjs
```

### What is the expected behavior?

The script should simply emit this to the console:

```sh
{
  "extends": 1
}
```

### What do you see instead?

```sh
1 | { "extends": 1 }
    ^
SyntaxError: Cannot use the keyword 'extends' as a variable name.
      at <parse> (/Users/[...]/bug-repliaction/tsconfig.json:1:1)
```

### Additional information

If I rename the JSON file to`test.json` and update the import path, then the script runs just fine.

**Comment by adamjosefus** [comment]
Similar situation with "require". Relative path works, absolute path doesn't.

```js
const data = require("/__absolute_path__/tsconfig.json");
console.log(data);
```

```bash
╰─$ bun main.js                                                                                                                                                                                                               130 ↵
1 | {
2 |   "compilerOptions": {},
3 |   "extends": "./tsconfig.base.json",
    ^
SyntaxError: Cannot use the keyword 'extends' as a variable name.
      at <parse> (/__absolute_path__/tsconfig.json:3:1)
      at /__absolute_path__/main.js:2:7
```

**Comment by maranomynet** [comment]
This seems to be fixed in `bun@1.1.1` (possibly earlier, not sure)
